crates/evm/evm/src/executors/mod.rs

```diff
@@ -356,6 +374,16 @@ impl Executor {
     /// Commit the changeset to the database and adjust `self.inspector_config`
     /// values according to the executed call result
     fn commit(&mut self, result: &mut RawCallResult) {
+        // Persist factory deps from just executed tx
+        if let Some(zk_tx) = self.zk_tx.take() {
```

Comment:
This is connected to the above, but basically I don't think we should be editing it _during_ the call. 
I would consider this as some data in the DB, thus any changes should be persisted thru `commit`, which is what I replicated here. This is also the reason why above we make a clone.
To be honest, I noticed it when adding it in `call_raw`, where we only have `&self`, and where data is not commited. There are other methods that wrap both `call_raw` and `call_raw_with_env`, which will `commit` instead of these 2.
If we didn't have `&mut` needed later we could just pass a normal reference here instead.

Reply:
> I don't think we should be editing it during the call.
I think we have two situations where factory_deps are persisted:
1. The `CheatcodeTracer`, where it has no notion of `DatabaseCommit` and hence no need for this separation of concern. Any operation is applied to `journaled_state` which is then consumed to obtain the `state_diff`.
2. The executor, where we _do_ have this separation, and makes sense to write only on commit. Hence it should be carefully thought through.

`call_raw` is used during fuzzing where no commits should occur as the state is copy-on-write, `call_raw_with_env` is used for everything else. 

Additionally, it seems that we already have a reference to the `InspectorStack` in `call_raw_with_env` which uses the `CheatcodeTracer` that already contains these persistent factory deps - think it would be easier to simply copy over the cloned `inspector.cheatcodes.persisted_factory_deps` to `self.inspector.cheatcodes.persisted_factory_deps` and that avoids having another field.